### ISSUE TITLE
JAX batched cell list can miss neighbors with static max_total_cells

### ISSUE THREAD

**Issue by jonkhler** [NVIDIA/nvalchemi-toolkit-ops#128]
### Environment

- nvalchemi-toolkit-ops 0.4.0
- JAX 0.11.0
- Warp 1.15.0
- CUDA GPU

### Problem

`batch_build_cell_list` requires `max_total_cells` inside `jax.jit`, but when that argument is supplied it initializes `neighbor_search_radius` to ones instead of deriving it from the generated `cells_per_dimension` and cell face distances. The returned radius can therefore be too small and `batch_cell_list` / `batch_query_cell_list` silently omit valid neighbors.

For an identity cell, cutoff 0.3, and non-periodic random points, the builder returns `cells_per_dimension = [6, 6, 6]` but `neighbor_search_radius = [1, 1, 1]`. Since each bin is 1/6 wide, a radius of 2 is required for a 0.3 cutoff. In a 128-query / 128-key test using the public `target_indices` path, a brute-force reference found 1329 bipartite pairs while the cell list returned 1077.

### Workaround / expected behavior

Calling public `batch_build_cell_list`, replacing the radius with

```python
face = 1 / jnp.linalg.norm(jnp.linalg.inv(cell).transpose(0, 2, 1), axis=2)
radius = jnp.ceil(jnp.nextafter(cutoff * cells_per_dimension / face, jnp.inf)).astype(jnp.int32)
radius = jnp.where((cells_per_dimension == 1) & ~pbc, 0, radius)
```

and then calling public `batch_query_cell_list` restores exact edge-set equality with brute force and kUPS for all eight 3D PBC masks.

The `max_total_cells is None` branch calls the estimator and receives a derived radius, but that branch is documented as not JIT-compatible. The static-capacity branch should derive the radius after constructing the actual bin counts as well.
